proposals/accepted/051-ghc-base-libraries.rst

```diff
@@ -0,0 +1,330 @@
+  (But CLC would not expect to curate the API of ``ghc-experimental``.)
+
+- Perhaps ``ghc-experimental`` should be in the purview of the GHC Proposals process.
+  GHC devs should not just make up random APIs and pop them into ``ghc-experimental``; a scrutiny process would be valuable.
+
+- Under this proposal, there is initially no change (whatsoever) to the API exposed by ``base``, or its performance characteristics.
+  The impact on clients should therefore be zero.
+
+  Over time, the GHC developers may make CLC proposals to remove types and functions that are currently in the ``base`` API, but are in truth part of GHC's implementation, and were originally exposed by historical accident.
+  But these are *future*\ proposals.
+
+  To make the transition suggested in these future proposals easier to manage, we have in progress a "deprecated exports" mechanism that will ease such transitions.
+  For a transitional period, ``base`` can continue to export the function, but with a deprecation warning saying something like:
+
+    This is going to disappear from base.
+    You probably don't want to use it at all.
+    But if you absolutely must, get it from ``ghc-internals``.
+
+- To expose a new function from ``ghc-internals`` requires that any functions on which it depends are also in ``ghc-internals`` (not base).
+  So we may need to move code from ``base`` to ``ghc-internals``, leaving a shim behind in base.
+  In practice, that may mean that quite a lot of code will move into ``ghc-internals`` quite quickly.
+  But that's fine: *it is just an implementation matter*: provided the modules, exports, and API of ``base`` are maintained, it is immaterial to clients (and hence to CLC) exactly *how* they are maintained.
+
+- This proposal is fully compatible with, and actively supports, the `CLC charter <https://github.com/haskell/core-libraries-committee#base-package>`__:
+
+    The primary responsibility of CLC is to manage API changes of ``base`` package.
+    The ownership of ``base`` belongs to GHC developers, and they can maintain it freely without CLC involvement as long as changes are invisible to clients.
+    Documentation changes normally fall under GHC developers purview, except significant ones (e.g., adding or changing type class laws).
+
+- It also supports GHC innovation, by
+
+  - allowing GHC freedom to change aspects of its implementation
+
+  - allowing the GHC Steering Committee to add new functions and types in ``ghc-experimental``.
+
+- One might wonder why GHC has three "internal" packages: ``ghc-internals``, ``ghc-bignum``, and ``ghc-prim``? Could they not be a single package? Answer: technically yes, but it helps to keep dependencies and responsibilities clear.
+  And it's purely an internal GHC matter; if the team wants to structure GHC's internals with three packages, or ten, that's up to them.
```

Comment:
Let's change this awkward phrasing:

```suggestion
- The three "internal" packages: ``ghc-internals``, ``ghc-bignum``, and ``ghc-prim``, could arguably be a single package, but the GHC team has decided that encapsulating the relevant code in this way helps to keep dependencies and responsibilities clear.
```

Reply:
I am fond of the second sentence, because its perhapps the only indication that `ghc-internals`'s very existence is also unstable; this follows from the GHC devs being allows to shuffle around code amongst their unstable libraries as often as they please.

Reply:
@mixphix's rephrasing of the first sentence is helpful. (I would remove "arguably" or replace with "in principle".)  But as John says, let's retain "It's purely an internal GHC matter"

Reply:
```suggestion
- The three "internal" packages: ``ghc-internals``, ``ghc-bignum``, and ``ghc-prim``, could arguably be a single package, but the GHC team has decided that encapsulating the relevant code in this way helps to keep dependencies and responsibilities clear.
  And it's purely an internal GHC matter; if the team wants to structure GHC's internals with three packages, or ten, that's up to them.
```